### ISSUE TITLE
Searching poe.trade causes AHK to become temporarily unresponsive 

### ISSUE THREAD

**Issue by corycomer** [PoE-TradeMacro/POE-TradeMacro#291]
When in game, if I hover over an item and hit CTRL+C I can see the tooltip just fine and it appears instantly. However when I hit CTRL+D the tooltip comes up indicating it is searching, but then when I check the script in the system tray AHK will show "Not Responding" for about 30-45 seconds or longer at times. After that, it does eventually fetch the results and display them in the tooltip. This happens about 60-70% of the time, but there are times that the response comes back almost immediately and updates the tooltip normally like I'd expect. 

This is on Windows 10 btw. I've tried running both Run_TradeMacro.ahk and Run_TradeMacro_no_Admin_Elevation.ahk and I've tried running both as Administrator. I've also tried setting the compatibility mode of the AutoHotkey binary to run as Administrator thinking maybe there was a permissions issue but that didn't help. It seems almost like something is timing out during searches and then on a retry it is getting the response maybe?

Any suggestions on what to look for, test, or settings to change this would be greatly appreciated. I end up going back to using the site most of the time because the responses just take too long when trying to quickly determine if something is of any worth :/

Edit: Oh, I should also mention that once it does get a response back the first time, subsequent searches for a while respond without any delay. So the first search can take ~30-60 seconds or more, and then searches after that only take 1-3 seconds. Maybe something is being cached? Does this possibly have to do with cookies?

**Comment by Eruyome** [comment]
The request uses WinHttp, I'm not aware of any connection retries, but this involves a small redirect. After sending the POST request with all search parameters to poe.trade it redirects to the actual search result page. This redirect is being skipped when you open the search on poe.trade (ctrl + q). 

That leaves manual retires (doing a search a second time). It wouldn't be weird to have different response times but yours seem to be a bit to long and not random enough. The request doesn't send any cookies (so, shouldn't use any) since poe.trade doesn't have cloudflare protection atm.

In general the connection can be bad/slow because of your router in which case a reboot can help.. Had this too.

As long as the search is simply slow but doesn't fail I don't really know what could be wrong aside from issues with your local network/pc. Maybe I can think of something tomorrow, it's pretty late.

**Comment by corycomer** [comment]
I have a suspicion now that the issue might be with Windows 10 itself. I wanted to grab something from the stash API so wrote a quick python script to grab the response and dump it to the console, and I noticed that the requests module in python was also taking a really long time. I ran -m cProfile and it looks like there's definitely an issue with making requests. I wonder if maybe there's some kind of DNS resolution issue because with the TradeMacro, the first search takes a while, but then I can search 10-20 times afterwords and it works almost instantly, but if I wait 5-10 minutes before doing another search, that next search takes a long time again, so that smells like DNS caching to me. FireFox and Chrome work fine, but they also both maintain their own DNS cache and handle resolution differently than the OS, so I'm going to dig into that when I have some time and post the results here in case anyone else experiences this issue. 

Thanks for taking a look, hopefully this is a quick issue to fix, as using the TradeMacro has changed the way I play considerably and I would love to use it even more!

**Comment by Eruyome** [comment]
Good luck on your search then.

**Comment by corycomer** [comment]
Just wanted to follow up with what I found. After some testing and profiling I noticed that anything using Windows DNS resolution was taking a long time, and it seems like the reason was because of an issue with some changes that were made to the way DNS resolution happens when you have multiple interfaces. A number of IPv6 lookups were failing or returning the wrong information (likely due to a VPN connection I have) and resulting in timeouts and failed connections. To solve the issue, I disabled IPv6 completely on the interface and disabled "smart" DNS resolution. 

Since then my searches through the TradeMacro have all taken 2-3 seconds tops! 

Here's some references for anyone who might experiencing a similar issue:

https://support.microsoft.com/en-us/help/929852/how-to-disable-ipv6-or-its-components-in-windows
https://superuser.com/questions/969171/multihomed-windows-10-dns-resolution-timeouts

Hope this helps anyone else who might be having a similar issue!
